brackend/endpoints/tournaments/tournaments.py

```diff
@@ -1,36 +1,38 @@
-from flask import Blueprint, request, jsonify
+from flask import Blueprint, current_app, jsonify, request
+from flask_restful import Api, Resource
 
-from flask_restful import Api
-from flask_restful import Resource
+from brackend.tasks.auth import auth_decorator
+from brackend.tasks.tasks import get_tournament_by_id, save_new_tournament, get_tournaments_by_uid
 
-from brackend.tasks.tasks import (
-    save_new_tournament,
-    get_tournament_by_id,
-)
-
-tournament_bp = Blueprint('tournaments', __name__)
+tournament_bp = Blueprint("tournaments", __name__)
 tournament_api = Api(tournament_bp)
 
 
 class Tournaments(Resource):
```

Comment:
Note, I thought you should be able to do:
```suggestion
@auth_decorator
class Tournaments(Resource):
```

and remove `method_decorators = [auth_decorator]` - I find it much easier to tell what decorators are on an endpoint with that, but wondering why I don't see this as an example in the docs

Reply:
This took me down a fun python class rabbit hole